core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -944,6 +944,14 @@ public XContentBuilder percentageField(String rawFieldName, String readableField
         return this;
     }
 
+    public XContentBuilder nanosAsMillis(String rawFieldName, String readableFieldName, long nanos) throws IOException {
```

Comment:
I don't think so. TimeUnit#toString doesn't seem to be used for precise output, but I also don't think we should change that because it is certainly used all over the place. If we are fine with reducing the profile output precision for the human readable format we can use TimeUnit, otherwise we should stick to the current 10-digit-ms format.